lib/classes/board.rb

```diff
@@ -0,0 +1,77 @@
+module OurTicTacToe
+	class Board
+		attr_reader :grid
+		def initialize(grid_board = {})
+			@grid = grid_board.fetch(:grid, default_board)
+		end
+
+		def get_cell(row, col)
+			grid[row][col]
+		end
+
+		def set_cell(row, col, value)
+			cell = get_cell(row, col)
+			cell.value.empty? ? cell.value = value : nil
+		end
+
+		def game_over
+			return :winner if winner?
+			return :draw if draw?
+
+			false
+		end
+
+		def formatted_grid
+			grid.each_with_index do |row, outer|
+				r = row.each_with_index.map do |col, inner|
+					if col.value.empty?
+						case outer
+						when 0
+							inner + 1
+						when 1
+							inner + 4
+						when 2
+							inner + 7
+						end
+					else
+						col.value
+					end
+				end
+				puts r.join(' ')
+			end
+		end
+
+		def winning_positions
+			grid + grid.transpose + diagonals
+		end
```

Comment:
Not bad *per se* but we are generating a whole new matrix by calling `grid.transpose` just to make the checks. You can leave it like this, but I want you to think about performance when building apps. Do we really need to generate a whole new matrix just to check for the columns?

Life is full of tradeoffs. In programming it may come in the form of commodity vs performance.

Reply:
Okay great, thank.

We would create an issue out of this and see if we can address it at a later date. 

Thanks again. See iss #13